inc/ILogOutputter.hpp

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include "LogData.hpp"
+
+namespace weblog {
+
+/**
+ * @brief Interface for log outputters.
+ *
+ * The ILogOutputter interface is used to implement different log outputters.
+ * The log outputters must implement the log function to log messages and the
+ * getFormattedMessage to format the message.
+ * The copy ctor and copy assignment operator are implemented to silence clang-tidy warnings.
+ */
+class ILogOutputter {
+
+public:
+	ILogOutputter() {};
+	virtual ~ILogOutputter() {};
+
+	virtual void log(const LogData& logData) = 0;
+
+private:
+	ILogOutputter(const ILogOutputter& ref) { (void)ref; };
+	ILogOutputter& operator=(const ILogOutputter& rhs)
+	{
+		if (this == &rhs)
```

Comment:
These should not be used anyway. These implementations exist to silence warnings: 
- Copy assignment is implemented since I implement a virtual destructor
- if (this == &rhs) is written since it checks for self-assignment
- rhs is voided since it is unused.

Alternative would be to //NOLINT this interface, but prefer to be explicit.

Reply:
I see you changed the whole thing from interface to an abstract class. And moved the definition into the .cpp file. But what I mentioned originally still matters. See below.